packages/loot-core/src/server/accounts/transfer.js

```diff
@@ -101,12 +125,24 @@ export async function removeTransfer(transaction) {
       // would invalidate the whole split transaction. Instead of turn
       // it into a normal transaction
       await db.updateTransaction({
-        id: transaction.transfer_id,
+        id: transferTrans.id,
         transfer_id: null,
         payee: null,
       });
+    } else if (transferTrans.cleared) {
+      // do not delete related transaction if cleared just null the transfer_id and reset payee
+      let { id: fromPayee } = await db.first(
+        'SELECT id FROM payees WHERE name = ?',
+        [transaction.imported_payee],
+      );
+
+      await db.updateTransaction({
+        id: transferTrans.id,
+        transfer_id: null,
+        payee: fromPayee || null,
```

Comment:
Maybe this could run rules again to select an appropriate payee after trying this fallback?